cpp/dolfinx/io/XDMFFile.cpp

```diff
@@ -336,8 +337,29 @@ XDMFFile::read_meshtags(const mesh::Mesh<double>& mesh, std::string name,
 
   const auto [entities, eshape] = read_topology_data(name, xpath);
 
-  pugi::xml_node values_data_node
-      = grid_node.child("Attribute").child("DataItem");
+  pugi::xml_node attribute_node = grid_node.child("Attribute");
+  pugi::xml_node values_data_node = attribute_node.child("DataItem");
+  if (not attribute_label.empty())
+  {
+    bool found = false;
+    for (; attribute_node;
+         attribute_node = attribute_node.next_sibling("Attribute"))
+    {
+      pugi::xml_attribute hint;
```

Comment:
What is this for?

Reply:
It's something that `pugixml` requires, like a hint to try speed-up the search of a node.

Reply:
I don't think it can because `hint` is also an output parameter of the function.